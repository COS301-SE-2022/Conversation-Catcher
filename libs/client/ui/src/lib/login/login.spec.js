import React from 'react';
import { render } from '@testing-library/react-native';
import Login from './login';
import { create } from 'react-test-renderer';

describe('Login', () => {
  it('should render successfully', () => {
    const { container } = render(<Login />);
    expect(container).toBeTruthy();
  });

  /*
  const tree = create(<Login />);
  test('snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
  */
});
