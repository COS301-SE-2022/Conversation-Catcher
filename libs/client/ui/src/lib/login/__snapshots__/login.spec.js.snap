// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffffff",
      "flexDirection": "column",
      "flexGrow": 1,
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "12%",
        "justifyContent": "center",
        "minHeight": 28,
        "padding": 5,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#344053ff",
          "fontFamily": "System",
          "fontSize": 24,
          "fontStyle": "normal",
          "fontWeight": "700",
          "letterSpacing": 0,
          "lineHeight": 28,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "textAlign": "center",
        }
      }
    >
      Log in to your account
    </Text>
  </View>
  <View
    style={
      Object {
        "width": "85%",
      }
    }
  >
    <View
      style={
        Object {
          "padding": 7,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "flex-start",
            "paddingBottom": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#344053ff",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "500",
              "letterSpacing": 0,
              "lineHeight": 20,
              "paddingBottom": 5,
              "textAlign": "left",
            }
          }
        >
          Email
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffffff",
            "borderColor": "#d0d5ddff",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "elevation": 2,
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.2173913043478261,
            "shadowRadius": 2.621621621621622,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexGrow": 1,
              "justifyContent": "center",
              "margin": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingHorizontal": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 15,
                  },
                  Object {
                    "color": "#3F89BE",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <TextInput
            placeholder="johnsmith@gmail.com"
            style={
              Object {
                "color": "#344053ff",
                "flex": 1,
                "fontFamily": "System",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 20,
                "paddingHorizontal": 0,
                "paddingVertical": 0,
                "textAlign": "left",
              }
            }
            underlineColorAndroid="transparent"
          />
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 16,
                "marginBottom": 10,
                "marginLeft": 0,
                "marginRight": 5,
                "marginTop": 10,
                "minWidth": 16,
                "opacity": 1,
                "resizeMode": "contain",
                "width": 16,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 17,
                  },
                  Object {
                    "color": "#d0d5ddff",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "flex-start",
            "margin": 5,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "padding": 7,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "flex-start",
            "paddingBottom": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#344053ff",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "500",
              "letterSpacing": 0,
              "lineHeight": 20,
              "paddingBottom": 5,
              "textAlign": "left",
            }
          }
        >
          Password
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffffff",
            "borderColor": "#d0d5ddff",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "elevation": 2,
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.2173913043478261,
            "shadowRadius": 2.621621621621622,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexGrow": 1,
              "justifyContent": "center",
              "margin": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingHorizontal": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 21,
                  },
                  Object {
                    "color": "#3F89BE",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <TextInput
            placeholder="*******************"
            style={
              Object {
                "color": "#344053ff",
                "flex": 1,
                "fontFamily": "System",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 20,
                "paddingHorizontal": 0,
                "paddingVertical": 0,
                "textAlign": "left",
              }
            }
            underlineColorAndroid="transparent"
          />
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 16,
                "marginBottom": 10,
                "marginLeft": 0,
                "marginRight": 5,
                "marginTop": 10,
                "minWidth": 16,
                "opacity": 1,
                "resizeMode": "contain",
                "width": 16,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 17,
                  },
                  Object {
                    "color": "#d0d5ddff",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "flex-start",
            "margin": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#667084ff",
              "fontFamily": "System",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 20,
              "textAlign": "left",
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#3f89beff",
        "borderColor": "#3f89beff",
        "borderRadius": 8,
        "borderStyle": "solid",
        "borderWidth": 1,
        "elevation": 2,
        "height": "10%",
        "margin": 20,
        "opacity": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.2173913043478261,
        "shadowRadius": 2.621621621621622,
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffffff",
            "fontFamily": "System",
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 24,
            "paddingHorizontal": 0,
            "paddingVertical": 0,
            "textAlign": "left",
          }
        }
      >
        Log in
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#ffffffff",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "flexShrink": 1,
          "justifyContent": "flex-start",
          "margin": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#667084ff",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 16,
            "textAlign": "left",
            "textDecorationLine": "underline",
          }
        }
      >
        Forgot your password?
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#ffffffff",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "flexShrink": 1,
          "justifyContent": "flex-start",
          "margin": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#667084ff",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 16,
            "textAlign": "left",
            "textDecorationLine": "underline",
          }
        }
      >
        Don’t have an account?
      </Text>
    </View>
  </View>
</View>
`;
